mitodl/mitxpro
# Upgrade Webpack from v4 to v5

### What are the relevant tickets?
[#5077](https://github.com/mitodl/hq/issues/5077) 

### Description (What does it do?)
This PR upgrades dependencies related to webpack upgrade 5 to support python 3.12.x later.

### Screenshots (if appropriate):
**Before**:
![image](https://github.com/user-attachments/assets/74cecafd-e10c-4ae5-a5e8-e920c19d8e71)


**after:**
<img width="1278" alt="image" src="https://github.com/user-attachments/assets/b41fe277-5111-4084-8546-7c06161a5842">

### How can this be tested?

1.  Run `docker-compose up`. Your watch container will automatically generate the Webpack bundles for you.
2. Verify that the Webpack build is generated successfully with a webpack-stats.json file and that all xPro pages are working correctly.
3. Also, verify/test with the production configuration if possible.

**How to generate the bundles?**
1.  Run `docker-compose up`. Your watch container will automatically generate the Webpack bundles for you.
2.  To verify Webpack bundles for production, go to the watch container's shell and manually run `node node_modules/webpack/bin/webpack.js --config webpack.config.prod.js --bail`. This will generate the production bundles for you.
3. Use the `report.html` file to compare the differences visually.

Performance evalution:
- Switch between the master and this branch for testing.
- Generate the Webpack bundles on the master branch and note the bundle size.
- Generate the Webpack bundles on this branch and note the bundle size again.

### Additional Context

- After upgrading to Webpack 5, the `common` bundle along with a new one will automatically get pulled in with an additional `render_bundle` in the appropriate views.
